### PR TITLE
Header

### PR DESCRIPTION
**Pull request by Krank3n**
Added functional Header and Sidebar

### CODE REVIEW

File: src/js/components/Header/HeaderItems.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { CHeaderNavItem, CHeaderNavLink } from '@coreui/react';
+import CIcon from '@coreui/icons-react';
+import { FormattedMessage } from 'react-intl';
+import NavData from './NavData';
+
+const HeaderItems = () => {
```
Comment (jakxnz) [inline src/js/components/Header/HeaderItems.js:7]: Hey, I notice that these components are in the `Header` namespace (the `Header` folder). Can we move them to their own folders?

File: src/js/components/Header/SubHeader.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { CBreadcrumbRouter, CHeader, CSubheader } from '@coreui/react';
+import React from 'react';
+import NavData from './NavData';
+
+function SubHeader() {
+  return (
+    <>
+      <CSubheader className="px-3 justify-content-between">
```
Comment (jakxnz) [inline src/js/components/Header/SubHeader.js:8]: Do we wanna clean this up a bit?

File: src/js/components/Header/header.stories.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+export default {
+  title: 'Header',
+  component: Header,
+};
+
+export function HeaderStory() {
+  return (
+    <Header />
+  );
+}
```
Comment (jakxnz) [inline src/js/components/Header/header.stories.js:13]: Is this needed?

File: src/js/components/Header/HeaderNavigation.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  CHeaderBrand, CHeaderNav, CHeaderNavLink, CImg,
+} from '@coreui/react';
+import React from 'react';
+import Logo from '../../../images/logo.svg';
+import HeaderItems from './HeaderItems';
+
+function HeaderNavigation() {
+  return (
+    <>
+      <CHeaderBrand className="px-3 d-md-down-none">
```
Comment (jakxnz) [inline src/js/components/Header/HeaderNavigation.js:11]: Can we clean this up a bit?

File: src/js/components/Header/Header.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import {
+  CHeader,
+  CToggler,
+} from '@coreui/react';
+import HeaderNavigation from './HeaderNavigation';
+import SubHeader from './SubHeader';
+
+import ViewContext from './ViewContext';
+
+function Header() {
```
Comment (jakxnz) [inline src/js/components/Header/Header.js:11]:
```suggestion

import ViewContext from './ViewContext';

const Header = () => {
```

File: src/js/components/Header/header.stories.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
```
Comment (jakxnz) [inline src/js/components/Header/header.stories.js:1]: Could we move this to `src/stories/components/Header/`?

**Review by jakxnz — CHANGES_REQUESTED** [review]
Mmmmf, loving this progress.

File: src/js/components/Header/SubHeader.js (hunk 2 above)
Comment (Krank3n) [reply src/js/components/Header/SubHeader.js:8]: done

File: src/js/components/Header/header.stories.js (hunk 6 above)
Comment (Krank3n) [reply src/js/components/Header/header.stories.js:1]: done

File: src/js/components/Header/header.stories.js (hunk 3 above)
Comment (Krank3n) [reply src/js/components/Header/header.stories.js:13]: Yes it is... Storybook wont pick it up without it.

File: src/js/components/Header/Header.js (hunk 5 above)
Comment (Krank3n) [reply src/js/components/Header/Header.js:11]: I swear I tried this, and I couldnt nest another const in it. maybe it didnt have the () =>

Comment (Krank3n) [reply src/js/components/Header/Header.js:11]: anyway added thanks dude

File: src/js/components/Header/HeaderItems.js (hunk 1 above)
Comment (Krank3n) [reply src/js/components/Header/HeaderItems.js:7]: Sub folders?

Comment (Krank3n) [reply src/js/components/Header/HeaderItems.js:7]: Broken them to its own folders

File: src/js/components/Header/SubHeader.js (hunk 2 above)
Comment (Krank3n) [reply src/js/components/Header/SubHeader.js:8]: cleaned

**Review by jakxnz — CHANGES_REQUESTED** [review]
A global header with navigation ✅

image
image
image

Routing works:
image

The only issue is the images aren't being referenced in the correct location:
image

( The `src` above should be `dist/1089...`) 

Webpack seems to be dumping the dist images in the `dist/` root, can we output them to `dist/images/`?
image

I also notice that there's a duplicate `index.html` file being generated in the `dist/` folder, can we avoid that somehow?

We also need CI to be passing before we can merge.

**Review by jakxnz — CHANGES_REQUESTED** [review]
Doing some quick functional tests: I notice that when I use the desktop nav, the app routes as expected, but when I use the mobile nav the app doesn't seem to route and instead the browser just redirects (which then breaks the view, because there's no response - just a `404` - for the URL it redirects to)

image

**Comment by Krank3n** [comment]
Thanks


> Doing some quick functional tests: I notice that when I use the desktop nav, the app routes as expected, but when I use the mobile nav the app doesn't seem to route and instead the browser just redirects (which then breaks the view, because there's no response - just a `404` - for the URL it redirects to)
> 
> image

Hey man, thanks for finding that, should be all good now.

**Review by jakxnz — APPROVED** [review]
LGTM

[](https://tenor.com/view/button-press-the-button-button-mashing-bulb-boy-flip-the-switch-gif-19815853)

Note: The routing isn't working for each sub-url, (i.e navigating directly to `/resorts`) but I don't think that's in-scope for this story, we can remedy that when we focus more on routing.

**Pull request merged**